docs/advanced_topics.rst

```diff
@@ -48,6 +48,7 @@ The purpose of this functionality is to enable the user to fine-tune the process
 DALI uses intermediate buffers to pass data between operators in the processing graph. With DALI, the memory is never freed but just enlarged when present buffers are not sufficient to hold the data. However, in some cases, even this limited number of allocations still could affect DALI performance. Hence, if the user knows how much memory each operator buffer needs, then it is possible to provide a hint to presize buffers before the first run.
 Two parameters are available: First, the ``bytes_per_sample`` pipeline argument, which accepts one value that is used globally across all operators for all buffers.
 The second parameter is the ``bytes_per_sample_hint`` per operator argument, which accepts one value or a list of values. When one value is provided it is used for all output buffers for a given operator. When a list is provided then each buffer is presized to the corresponding size.
+To learn how much memory outputs of each operator need, the user may create the pipeline with ``enable_memory_stats`` set to ``True`` and then query the pipeline for the operator's output memory statistics by calling ``executor_meta`` method on the pipeline. The ``max_real_memory_size`` value tells what is the biggest tensor in the batch for the outputs that allocate memory per sample, not for the whole batch at the time, or average tensor size when the allocation is continuous. This value is the one that should be provided to ``bytes_per_sample_hint``.
```

Comment:
Shouldn't that be the `max_reserved_memory_size` as this is the bigger value which includes some "padding"? Or, as we calculate max, both will be similar here (and we don't need to actually return all the values?) When they will be different - only for small data with the padding?

Reply:
Not really. If the user uses `SetBufferGrowthFactor` with a big value then better stick to the used memory, not the reserved one.

Reply:
Ok, it makes more sense to use the real, as there won't be padding used if there is enough memory -> no reallocation.